cli/src/main/java/org/jboss/as/cli/handlers/ifelse/ExpressionParser.java

```diff
@@ -138,6 +138,8 @@ protected Operand parseOperand() throws CommandLineException {
                     } else {
                         --depthCount;
                     }
+                } else if (input.charAt(endIndex) == '"') { // traverse over string operands surrounded by quotes
```

Comment:
@TomasHofman this issue is not that simple. We shouldn't be adding escaping here at all. When the CLI receives an input, it parses the line which already includes processing of escaped characters and resolution of expressions (system properties and variables). The command handler then receives arguments already resolved (with escaping processed too). Unresolved argument value will still be available to the handler too.

So, if we introduce character escaping here, it'll be the second one already. On the other hand, if we ask for the unresolved argument value then we also have to take into account that the line might contain unresolved expressions.

This commit unfortunately doesn't solve the general issue. E.g.
if (outcome == success && result.value != "a64180c0)e7187af86b435229de904104\"") of /system-property=test:read-resource

In the line above my intention was to include the quote into the value using \. But the condition string parsed by ExpressionParser will be (outcome == success && result.value != "a64180c0)e7187af86b435229de904104"")

I wish we could just take your commit in and be done with it but we need a proper parser here. Sorry but thanks for bringing this up!